### ISSUE TITLE
If an admin user is created set root's password to an exclamation mark

### ISSUE THREAD

**Issue by pdxjohnny** [clearlinux/clr-installer#212]
**Describe the bug**
Currently, if an admin user (account with sudo access) is created, `root` user is still asked to set its password the first time it logs in.

**Expected Behavior**
The user should not need to lock out the `root` account manually after install. If an admin account is created, the installer should also lock out the `root` account.

**References**
One way lock out an account is by setting the `encrypted password` field of the `/etc/shadow` file to `!`.

```
SHADOW(5)                             File Formats and Conversions                            SHADOW(5)

       encrypted password
           Refer to crypt(3) for details on how this string is interpreted.

           If the password field contains some string that is not a valid result of crypt(3), for
           instance ! or *, the user will not be able to use a unix password to log in (but the user
           may log in the system by other means).

           This field may be empty, in which case no passwords are required to authenticate as the
           specified login name. However, some applications which read the /etc/shadow file may decide
           not to permit any access at all if the password field is empty.

           A password field which starts with an exclamation mark means that the password is locked.
           The remaining characters on the line represent the password field before the password was
           locked.
```

**Additional context**
Setting `root` encrypted password to `!` in `/etc/shadow` does indeed disable the new password prompt on login. Tested on Clear Linux version 27700.

**Comment by mvincerx** [comment]
Glad that you opened this. I've been curious about this default behavior after revising bare-metal-install.

**Comment by mdhorn** [comment]
Just to be clear on the requirement, if a user account is created AND it is flagged as an admin account, the root account will be disabled for password login.

Creating accounts with none set to admin, or only having the default root account will remain the same as Clear Linux as always bee - the first login to the root account prompts for setting a password.

**Comment by mdhorn** [comment]
The soon-to-come pull request locks the root account if any user accounts are created which have admin access -- expect in the case where the root user is specifically added (which can only be done from the command line as TUI dis-allows configuring any known system accounts).

After the installation, root will no longer prompt for a password as the account is locked:
```
clr-3b3a08b0501d40e5ab299a55d665b7ea login: root
Password:

Login incorrect
```

But sudo is working:
```
clr-3b3a08b0501d40e5ab299a55d665b7ea login: mhorn
Password:
mhorn@clr-3b3a08b0501d40e5ab299a55d665b7ea ~ $ sudo cat /etc/shadow
mhorn:$6$JyDFhAxd8L/4OxP$1oqFOCp5rWOjPixJjMn.zq/KsxOty699ie1UfPLUue3skC57GMDX.3Q5dWuBU5nOVzYv6urT./ax0FbCbH.U/:17939::::::
root:!:0:0:99999:7:::
```

**Comment by mdhorn** [comment]
**Updated Requirements:**
- If the root account is defined with an encrypted password in the YAML configuration file, do not lock the root account.
- If the root account is defined with SSH key(s) in the YAML configuration file, and there is no password set for the root account, then lock the root account.

**Comment by mdhorn** [comment]
Fixed by #213 in release 1.1.5
